ubiquity/dell-bootstrap.py

```diff
@@ -379,6 +387,9 @@ def disk_type_key(d):
         return d2_key - d1_key
 
 
+def is_boot_with_hdd_flag():
+    return 'dell-recovery/recovery_type=hdd' in open('/proc/cmdline', 'r').read().split()
```

Comment:
so does this mean that basically `dell-recovery/recovery_type=hdd` and `dell-recovery/recovery_type=dev` both get set on kernel command line?  If so, that's a level of complexity that I didn't really appreciate here.  It means we lose the one to one mapping of recovery types.  I was actually envisioning that `dell-recovery/recovery_type=dev` could be what passes around everywhere and that's all that is used.

Reply:
Don't get your point.

For now, per my way of using it
1. I preseed recovery_type in /preseed/dell-recovery.seed.
2. kernel parameter dell-recovery/recovery_type=hdd only appear in recovery from hdd
3. for recovery from hdd, even we set to dev type, it will still pass dell-recovery/recovery_type=hdd, not dell-recovery/recovery_type=dev.
   We will NOT see dell-recovery/recovery_type=dev with this PR.

Reply:
Can you share exactly what is in your `/preseed/dell-recovery.seed` to make it work?  And is my understanding right that you just put this on your USB key and then USB key is fully automated until at desktop with this change?

Reply:
I put

dell-recovery dell-recovery/recovery_type string dev